code/game/machinery/_machinery.dm

```diff
@@ -231,6 +234,36 @@ Class Procs:
 				return FALSE
 	return TRUE
 
+/obj/machinery/proc/check_nap_violations()
+	if(!SSgoldmansachs.full_ancap)
+		return
+	if(occupant && !state_open)
+		if(ishuman(occupant))
+			var/mob/living/carbon/human/H = occupant
+			var/obj/item/card/id/I = H.get_idcard()
+			if(I)
+				var/datum/bank_account/insurance = I.registered_account
+				if(!insurance)
+					say("[market_verb] NAP Violation: No bank account found.")
+					nap_violation()
+					return FALSE
+				else
+					if(!insurance.adjust_money(-1 * fair_market_price))
+						say("[market_verb] NAP Violation: Unable to pay.")
+						nap_violation()
+						return FALSE
+					var/datum/bank_account/D = SSgoldmansachs.get_dep_account(payment_department)
+					if(D)
+						D.adjust_money(fair_market_price)
```

Comment:
I'm not removing sanity checks in SS13. It's fucking SS13.

Reply:
you can have sanity checks without the deep as fuck nesting you know that

Reply:
i'm going to need you to explain what's actually wrong here instead of repeating cargo cult memes at me

Reply:
I'm talking about this entire part of code not just this specific tidbit, I just cant show more of it in 1 comment. Sorry shouldve clarified that but I wasn't sure how much of the code it would actually show

Reply:
oh, the entirety of this code is perfectly fine, I'm not wasting time processing anything that doesn't need to be processed and I have actual error messages depending on the result. It's not changing.

Reply:
ok